client/service/src/builder.rs

```diff
@@ -906,9 +906,9 @@ ServiceBuilder<
 			let network = Arc::downgrade(&network);
 			let transaction_pool_ = transaction_pool.clone();
 			let events = transaction_pool.import_notification_stream()
-				.for_each(move |_| {
+				.for_each(move |hash| {
```

Comment:
I'd like not to. 

We might handle those differently later. 

For example, accumulate some hashes and query all later or something else. I think unique identifier is a natural choice that will allow different scenarios later.

Also, notifications have clear api by value everywhere else (or should have).

Reply:
I've been discussing this with @NikVolf over DMs for quite a while, and we concluded that indeed it doesn't make much sense to pass references anyway. Major reasons:
1. The networking needs to clone the extrinsic before constructing the packet anyway (the packet owns extrinsic data).
2. We could clone early and then pass data through the channel message, but it would be inconsistent with existing APIs and doesn't really make much sense. Just extra clone in the thread that handles notifications instead of cloning only if we actually need the hash (i.e. we know that the other peer does not have it).

Hence it makes more sense to just pass `hash` to the networking and let the networking request transactions by-hash as it works currently in this PR.